server/app/src/test/java/com/google/heartrate/wearos/app/gatt/heartrate/characteristics/HeartRateGattCharacteristicManagerTest.java

```diff
@@ -0,0 +1,159 @@
+package com.google.heartrate.wearos.app.gatt.heartrate.characteristics;
+
+import android.os.Build;
+
+import com.google.heartrate.wearos.app.gatt.CharacteristicsArgumentProvider;
+import com.google.heartrate.wearos.app.gatt.GattException;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+import static com.google.heartrate.wearos.app.gatt.CharacteristicsArgumentProvider.Range.IN_UINT16;
+import static com.google.heartrate.wearos.app.gatt.CharacteristicsArgumentProvider.Range.IN_UINT8;
+import static com.google.heartrate.wearos.app.gatt.CharacteristicsArgumentProvider.Range.OUT_UINT16;
+import static com.google.heartrate.wearos.app.gatt.TestUtils.assertNotThrows;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(sdk = {Build.VERSION_CODES.O_MR1}, manifest = Config.NONE)
+public class HeartRateGattCharacteristicManagerTest {
+
+    private HeartRateMeasurementCharacteristic characteristic;
+
+    @Before
+    public void setUp() {
+        characteristic = new HeartRateMeasurementCharacteristic();
+    }
+
+    /**
+     * Assent setting values to given Heart Rate Characteristic was successful.
+     * <br/>Assents no exceptions was thrown and after set Heart Rate Value and Expended Energy Value
+     * are equal to requested.
+     *
+     * @param characteristic      characteristic to set value
+     * @param heartRateValue      heart rate value to set
+     * @param expendedEnergyValue expended energy value to set
+     */
+    public static void assertSetValuesSuccessful(HeartRateMeasurementCharacteristic characteristic,
+                                                 int heartRateValue, int expendedEnergyValue) {
+        final int[] currentHeartRateMeasurementValue = new int[1];
+        final int[] currentExpendedEnergyValue = new int[1];
+
+        assertNotThrows(() -> characteristic
+                .setHeartRateCharacteristicValue(heartRateValue, expendedEnergyValue));
+        assertNotThrows(() -> currentHeartRateMeasurementValue[0] = characteristic
+                .getHeartRateMeasurementValue());
+        assertNotThrows(() -> currentExpendedEnergyValue[0] = characteristic
+                .getExpandedEnergyValue());
+
+        assertEquals(heartRateValue, currentHeartRateMeasurementValue[0]);
+        assertEquals(expendedEnergyValue, currentExpendedEnergyValue[0]);
+    }
+
+    /**
+     * Assent set values to given Heart Rate Characteristic fails with GattException.
+     *
+     * @param characteristic      characteristic to set value
+     * @param heartRateValue      heart rate value to set
+     * @param expendedEnergyValue expended energy value to set
+     */
+    public static void assertSetValuesFailed(HeartRateMeasurementCharacteristic characteristic,
```

Comment:
private?

Reply:
I use this method in different tests. Don't want to copy-paste it.